### ISSUE TITLE
how to find the [distro] executable name?

### ISSUE THREAD

**Issue by skierpage** [MicrosoftDocs/WSL#267]
I installed Debian for WSL from the Windows Store. All the examples of configuring a distro in Windows use ubuntu(.exe). How can I find the Windows executable name of another installed distro? Things I've tried:`wslconfig /list /all` just reports "Debian (Default)"; I can't get properties of Debian from the Windows Store; if I max out a terminal, Resource Monitor reports the process running is just "conhost.exe".

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: deab370f-a7a1-ed3f-89ae-22194fd8fb78
* Version Independent ID: 9f406c77-415c-80db-d596-0a7fa4ea2ba9
* Content: [Manage Linux Distributions](https://docs.microsoft.com/en-us/windows/wsl/wsl-config)
* Content Source: [WSL/wsl-config.md](https://github.com/MicrosoftDocs/WSL/blob/live/WSL/wsl-config.md)
* Service: **windows-subsystem-for-linux**
* Product: **windows-subsystem-for-linux**
* GitHub Login: @scooley
* Microsoft Alias: **scooley**

**Comment by TotallyInformation** [comment]
Can't believe this hasn't been responded to.

With limited knowledge: I don't think that the distro is an executable. Rather it is a package that wsl.exe loads and runs. So you cannot see the distro that is running in the Task Manager.

As far as I can see, the executable name is simply the distro name listed by wslconfig. In your case, the distro name is "Debian" so the command to run it would be "debian.exe". However, as you only have a single distro, you can simply do wsl.exe since it is your default anyway.

conhost.exe is the container that runs command shells which is why it always (confusingly) appears in task manager.

**Comment by craigloewen-msft** [comment]
You can run specific distros from the command line by running `wsl -d <distroName>` and seeing a list of distro names with `wsl -l -v`.

WSL has gone through many updates and revisions, and this issue may be fixed in the latest version. We're cleaning out some of our older issues, and will mark this as closed. If you are still experiencing this issue on the latest version please feel free to reopen this! Thank you!

**Comment by N7K4** [comment]
I know that issue is closed since ages, Sorry for drop anyway an comment here.

What is the execute able name, when I have **multiple** same distros. In my case I have three ubuntu distro installed, via `--import` I can change the config for the FIRST one only. I have no idea how to get the names of the second and third.

my distro list

```
ubuntu
ubuntu-var1
ubuntu-var2
```
All distros are imported from the same exported file. I use the different distros to compare the behaviour.

**Comment by crramirez** [comment]
Hello @N7K4,

If you installed the distros via --import, there is no executable. They can be only invoked via wsl -d command. I recommend you to use Windows Terminal to invoke them.

What config do you want to change? The default user? Add this to the /etc/wsl.conf:

```
[user]
default=yourusername
```

Regards,
Carlos

**Comment by N7K4** [comment]
Hi @crramirez ,

Thanks for your responds. Yes I used --import. Thanks for your explaination, that there is no executeable.

Ragarding to, yes I found that also of a internet serach. 
> What config do you want to change? The default user? Add this to the /etc/wsl.conf:
>
> [user]
> default=yourusername

I did the changes, on two different ways, both did **not** work
1. From inside the WSL, I created the /ect/wsl.conf with touch and edited with vim (and saved it). After exit the terminal and come back, still root is my user not my configured one
2. Over Windows with the network \\wsl$ ... also no success

When I start wsl with -d and --user I can change the user, but only for that session. I am still not able to configure the user as new default. 

Any idea? Do I have to shutdown wsl or terminate the single instance of the wsl? Or is a simple exit and re-connect ok?

Many thanks, Nico
